src/test/java/org/commcare/formplayer/tests/MultiSelectCaseListTest.java

```diff
@@ -104,6 +103,23 @@ private void checkForSelectedEntitiesInstance(String sessionId, String[] selecti
         assertEquals(evaluateXpathResponseBean.getOutput(), result);
     }
 
+    @Test
+    public void testAutoAdvanceWithMultiSelect() throws Exception {
+        FormPlayerPropertyManagerMock.mockAutoAdvanceMenu(storageFactoryMock);
+        String[] selections = new String[]{"1"};
+        sessionNavigate(selections, APP,
+                EntityListResponse.class);
+
+        selections = new String[]{"1", "use_selected_values"};
+        String[] selectedValues =
+                new String[]{"5e421eb8bf414e03b4871195b869d894", "3512eb7c-7a58-4a95-beda-205eb0d7f163"};
+        NewFormResponse formResp = sessionNavigateWithSelectedValues(selections, APP, selectedValues,
+                NewFormResponse.class);
+
+        // selections should now be {"1", "guid"}
```

Comment:
this comment would be helpful in the `testAutoAdvanceMenuWithCaseSearch` test too